contracts/MoneyBoxUser.sol

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.20;
+
+import {MoneyBox} from "./MoneyBox.sol";
+
+abstract contract MoneyBoxUser {
+
+    function _calculateMoneyBoxAddress(address _creator, bytes32 _salt, address _token) internal pure returns (address) {
+
+        // From:
+        // https://docs.soliditylang.org/en/latest/control-structures.html#salted-contract-creations-create2
+        return address(uint160(uint(keccak256(abi.encodePacked(
```

Comment:
Please take a look here: https://docs.openzeppelin.com/contracts/5.x/api/utils#Create2
It's always nicer to use their code, as the audit comes for free.